app/src/ingest_bem_pdfs.py

```diff
@@ -0,0 +1,171 @@
+
+
+def to_markdown_texts(enriched_texts: list[EnrichedText]) -> list[EnrichedText]:
+    "Placeholder function. Will be implemented for DST-415, probably in a different file."
+    return enriched_texts
+
+
+@dataclass
+class ChunkTemp:
+    "Temporary class as we work through how we want to update document.py:Chunk"
+
+    # Minimal fields from Chunk
+    content: str
+    tokens: int
+    document: Document
+
+    ## Following are new fields
+
+    # For tracability back to a grouped_text
+    grouped_text_id: str  # The id from grouped_texts
+    parano: int
+    text_type: str  # "paragraph" or "list"
+
+    # Flatten 'page' data from grouped_texts
+    page_number: int
+
+    # Flattened 'headings' data from grouped_texts
+    # (heading level is apparent and heading pageno is not needed)
+    headings: list[str]
+    # ['SOME LEVEL 1 HEADING', 'Some Level 2 Heading', 'Oh my! A 3rd level heading']
+
+    # Number of splits (or chunks) the text was split into
+    num_splits: int  # = 1 (if not split)
+    # If not complete (num_splits > 1), specify the index starting from 0
+    split_index: int = 0
+
+
+def split_into_chunks(document: Document, grouped_texts: list[EnrichedText]) -> list[ChunkTemp]:
+    """
+    Given EnrichedTexts, convert the text to chunks and add them to the database.
+    """
+    chunks: list[ChunkTemp] = []
+    for paragraph in grouped_texts:
+        assert paragraph.id is not None
+        assert paragraph.page_number is not None
+
+        # For iteration 1, log warning for overly long text
+        embedding_model = app_config.sentence_transformer
+        token_count = len(embedding_model.tokenizer.tokenize(paragraph))
+        if token_count > embedding_model.max_seq_length:
+            logger.warning("Text too long for embedding model: %s", paragraph.text[:100])
+
+        text_chunks = [
+            # For iteration 1, don't split the text -- just create 1 chunk.
+            # TODO: TASK 3.a will split a paragraph into multiple chunks.
+            ChunkTemp(
+                content=paragraph.text,
+                tokens=len(paragraph.text.split()),
+                document=document,
+                grouped_text_id=paragraph.id,
+                parano=0,  # TODO
+                text_type=paragraph.type.name,
+                page_number=paragraph.page_number,
+                headings=[h.title for h in paragraph.headings],
+                num_splits=1,
+            )
+        ]
+        chunks += text_chunks
+    return chunks
+
+
+def _add_chunk(
+    db_session: db.Session, chunk_text: str, document: Document, current_token_count: int
+) -> None:
+    embedding_model = app_config.sentence_transformer
+    chunk_embedding = embedding_model.encode(chunk_text, show_progress_bar=False)
+    chunk = Chunk(
+        document=document,
+        content=chunk_text,
+        tokens=current_token_count,
+        mpnet_embedding=chunk_embedding,
+    )
+    db_session.add(chunk)
+
+
+def main() -> None:
```

Comment:
You'll want to also add this script to pyproject.toml and app/Makefile so it can be run from the command line for local development, like this: https://github.com/navapbc/labs-decision-support-tool/pull/31/files#diff-0c6c7defde945725f3d878fe20d0866dc29520f5f3816ff91ca56df2211e5451

Reply:
Added this task to https://navalabs.atlassian.net/browse/DST-419